src/ruleset_builder.hpp

```diff
@@ -100,7 +106,13 @@ class ruleset_builder {
     // Filters
     std::unordered_map<std::string_view, exclusion::rule_filter::ptr> rule_filters_;
     std::unordered_map<std::string_view, exclusion::input_filter::ptr> input_filters_;
-    // The list of targets used by rule_filters_, input_filters_ and their internal
+
+    // An mkmap organising scanners by their tags, used for processors
+    scanner_tag_map scanners_by_tags_;
```

Comment:
looks like this should be integrated into a structure combining `scanners_` and `scanners_by_tags_`, so that they are constructed/updated/destroyed in tandem. The current design is fragile. `scanners_by_tags_` has non-owning references to data inside `scanners_`, so invalidating `scanners_` will leave `scanners_by_tags_` dangling.

Reply:
Fair point, although do note that I'm planning to refactor the ruleset builder as it's quite hard to test as is now because so many things are going on inside it...